internal/internal_task_pollers.go

```diff
@@ -256,9 +252,19 @@ func (bp *basePoller) doPoll(
 	ctx, cancel, _ := newChannelContext(context.Background(), featureFlags, chanTimeout(pollTaskServiceTimeOut))
 
 	go func() {
+		defer close(doneC)
+		defer func() {
+			if p := recover(); p != nil {
+				bp.metricsScope.Counter(metrics.WorkerPanicCounter).Inc(1)
```

Comment:
We should have a dedicated metric here since WorkerPanicCounter is caused by panics in user-defined code, but this panic is caused by Cadence client internals.

Reply:
Done